### ISSUE TITLE
Toggling a token's invisibility does not fade out for the player

### ISSUE THREAD

**Issue by Fyorl** [foundryvtt/foundryvtt#7618]
Originally reported by ironmonk88#4075 https://discord.com/channels/170995199584108546/956306859491471420/1000940548783362198

**ALL MODULES DISABLED?** true  
**OS, Hosting, Browser (if applicable):** Windows 10, Chrome  
**Short Description of bug:** Toggling a tokens invisibility has a really nice fade for the GM.  But turns off instantly for the player.  I'm assuming that's because once invisible the token is hidden from the player.  It would be nice if the token could stay visible while there's a fade animation happening for the players.  Also, when it's fading back in for the player it looks like it starts from the GM's alpha instead of starting from 0.  
**Simple steps to reproduce the bug:** Create a Token.  Click on the HUD button to turn the token invisible.  It gradually fades for the GM.  But disappears for the player.

**Comment by Fyorl** [comment]
This might be a technical limitation of the vision mode visibility testing so I've not marked it as a bug if it's 'working as designed' for now.

**Comment by aaclayton** [comment]
Yes, it's not a bug, but this could potentially be a feature addition to treat a token as temporarily visible while it is in the process of hiding. This would be a fair bit of effort though so I think we shouldn't commit to this for V10.
